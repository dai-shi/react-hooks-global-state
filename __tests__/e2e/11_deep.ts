import puppeteer from 'puppeteer';

jest.setTimeout(15 * 1000);

describe('11_deep', () => {
  const port = process.env.PORT || '8080';

  it('should work with events', async () => {
    const browser = await puppeteer.launch();
    const page = await browser.newPage();
    await page.goto(`http://localhost:${port}/`);

    await page.click('div:nth-of-type(1) > button:nth-of-type(1)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.click('div:nth-of-type(1) > button:nth-of-type(2)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.click('div:nth-of-type(2) > button:nth-of-type(1)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.click('div:nth-of-type(2) > button:nth-of-type(2)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.type('div:nth-of-type(3) > input', '1');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.type('div:nth-of-type(4) > input', '2');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.type('div:nth-of-type(5) > div > input', '3');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.type('div:nth-of-type(6) > input', '4');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.type('div:nth-of-type(7) > input', '5');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.type('div:nth-of-type(8) > div> input', '6');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await browser.close();
  });
});
