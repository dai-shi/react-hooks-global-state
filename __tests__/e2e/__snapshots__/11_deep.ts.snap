// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`11_deep should work with events 1`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>(numRendered: 6)</span></div><div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>(numRendered: 8)</span></div><h1>Person</h1><div>First Name:<span></span><input value=\\"\\"><span>(numRendered: 2)</span></div><div>Last Name:<span></span><input value=\\"\\"><span>(numRendered: 4)</span></div><div><div>Age:<input value=\\"0\\"><span>(numRendered: 6)</span></div><div>Age Doubled:<span>0</span></div></div><div>First Name:<span></span><input value=\\"\\"><span>(numRendered: 8)</span></div><div>Last Name:<span></span><input value=\\"\\"><span>(numRendered: 10)</span></div><div><div>Age:<input value=\\"0\\"><span>(numRendered: 12)</span></div><div>Age Doubled:<span>0</span></div></div></div>
  

"
`;

exports[`11_deep should work with events 2`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>(numRendered: 10)</span></div><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>(numRendered: 12)</span></div><h1>Person</h1><div>First Name:<span></span><input value=\\"\\"><span>(numRendered: 2)</span></div><div>Last Name:<span></span><input value=\\"\\"><span>(numRendered: 4)</span></div><div><div>Age:<input value=\\"0\\"><span>(numRendered: 6)</span></div><div>Age Doubled:<span>0</span></div></div><div>First Name:<span></span><input value=\\"\\"><span>(numRendered: 8)</span></div><div>Last Name:<span></span><input value=\\"\\"><span>(numRendered: 10)</span></div><div><div>Age:<input value=\\"0\\"><span>(numRendered: 12)</span></div><div>Age Doubled:<span>0</span></div></div></div>
  

"
`;

exports[`11_deep should work with events 3`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>(numRendered: 14)</span></div><div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>(numRendered: 16)</span></div><h1>Person</h1><div>First Name:<span></span><input value=\\"\\"><span>(numRendered: 2)</span></div><div>Last Name:<span></span><input value=\\"\\"><span>(numRendered: 4)</span></div><div><div>Age:<input value=\\"0\\"><span>(numRendered: 6)</span></div><div>Age Doubled:<span>0</span></div></div><div>First Name:<span></span><input value=\\"\\"><span>(numRendered: 8)</span></div><div>Last Name:<span></span><input value=\\"\\"><span>(numRendered: 10)</span></div><div><div>Age:<input value=\\"0\\"><span>(numRendered: 12)</span></div><div>Age Doubled:<span>0</span></div></div></div>
  

"
`;

exports[`11_deep should work with events 4`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>(numRendered: 18)</span></div><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>(numRendered: 20)</span></div><h1>Person</h1><div>First Name:<span></span><input value=\\"\\"><span>(numRendered: 2)</span></div><div>Last Name:<span></span><input value=\\"\\"><span>(numRendered: 4)</span></div><div><div>Age:<input value=\\"0\\"><span>(numRendered: 6)</span></div><div>Age Doubled:<span>0</span></div></div><div>First Name:<span></span><input value=\\"\\"><span>(numRendered: 8)</span></div><div>Last Name:<span></span><input value=\\"\\"><span>(numRendered: 10)</span></div><div><div>Age:<input value=\\"0\\"><span>(numRendered: 12)</span></div><div>Age Doubled:<span>0</span></div></div></div>
  

"
`;

exports[`11_deep should work with events 5`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>(numRendered: 18)</span></div><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>(numRendered: 20)</span></div><h1>Person</h1><div>First Name:<span>1</span><input value=\\"1\\"><span>(numRendered: 14)</span></div><div>Last Name:<span></span><input value=\\"\\"><span>(numRendered: 4)</span></div><div><div>Age:<input value=\\"0\\"><span>(numRendered: 6)</span></div><div>Age Doubled:<span>0</span></div></div><div>First Name:<span>1</span><input value=\\"1\\"><span>(numRendered: 16)</span></div><div>Last Name:<span></span><input value=\\"\\"><span>(numRendered: 10)</span></div><div><div>Age:<input value=\\"0\\"><span>(numRendered: 12)</span></div><div>Age Doubled:<span>0</span></div></div></div>
  

"
`;

exports[`11_deep should work with events 6`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>(numRendered: 18)</span></div><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>(numRendered: 20)</span></div><h1>Person</h1><div>First Name:<span>1</span><input value=\\"1\\"><span>(numRendered: 14)</span></div><div>Last Name:<span>2</span><input value=\\"2\\"><span>(numRendered: 18)</span></div><div><div>Age:<input value=\\"0\\"><span>(numRendered: 6)</span></div><div>Age Doubled:<span>0</span></div></div><div>First Name:<span>1</span><input value=\\"1\\"><span>(numRendered: 16)</span></div><div>Last Name:<span>2</span><input value=\\"2\\"><span>(numRendered: 20)</span></div><div><div>Age:<input value=\\"0\\"><span>(numRendered: 12)</span></div><div>Age Doubled:<span>0</span></div></div></div>
  

"
`;

exports[`11_deep should work with events 7`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>(numRendered: 18)</span></div><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>(numRendered: 20)</span></div><h1>Person</h1><div>First Name:<span>1</span><input value=\\"1\\"><span>(numRendered: 14)</span></div><div>Last Name:<span>2</span><input value=\\"2\\"><span>(numRendered: 18)</span></div><div><div>Age:<input value=\\"3\\"><span>(numRendered: 22)</span></div><div>Age Doubled:<span>6</span></div></div><div>First Name:<span>1</span><input value=\\"1\\"><span>(numRendered: 16)</span></div><div>Last Name:<span>2</span><input value=\\"2\\"><span>(numRendered: 20)</span></div><div><div>Age:<input value=\\"3\\"><span>(numRendered: 24)</span></div><div>Age Doubled:<span>6</span></div></div></div>
  

"
`;

exports[`11_deep should work with events 8`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>(numRendered: 18)</span></div><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>(numRendered: 20)</span></div><h1>Person</h1><div>First Name:<span>14</span><input value=\\"14\\"><span>(numRendered: 26)</span></div><div>Last Name:<span>2</span><input value=\\"2\\"><span>(numRendered: 18)</span></div><div><div>Age:<input value=\\"3\\"><span>(numRendered: 22)</span></div><div>Age Doubled:<span>6</span></div></div><div>First Name:<span>14</span><input value=\\"14\\"><span>(numRendered: 28)</span></div><div>Last Name:<span>2</span><input value=\\"2\\"><span>(numRendered: 20)</span></div><div><div>Age:<input value=\\"3\\"><span>(numRendered: 24)</span></div><div>Age Doubled:<span>6</span></div></div></div>
  

"
`;

exports[`11_deep should work with events 9`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>(numRendered: 18)</span></div><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>(numRendered: 20)</span></div><h1>Person</h1><div>First Name:<span>14</span><input value=\\"14\\"><span>(numRendered: 26)</span></div><div>Last Name:<span>25</span><input value=\\"25\\"><span>(numRendered: 30)</span></div><div><div>Age:<input value=\\"3\\"><span>(numRendered: 22)</span></div><div>Age Doubled:<span>6</span></div></div><div>First Name:<span>14</span><input value=\\"14\\"><span>(numRendered: 28)</span></div><div>Last Name:<span>25</span><input value=\\"25\\"><span>(numRendered: 32)</span></div><div><div>Age:<input value=\\"3\\"><span>(numRendered: 24)</span></div><div>Age Doubled:<span>6</span></div></div></div>
  

"
`;

exports[`11_deep should work with events 10`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>(numRendered: 18)</span></div><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>(numRendered: 20)</span></div><h1>Person</h1><div>First Name:<span>14</span><input value=\\"14\\"><span>(numRendered: 26)</span></div><div>Last Name:<span>25</span><input value=\\"25\\"><span>(numRendered: 30)</span></div><div><div>Age:<input value=\\"36\\"><span>(numRendered: 34)</span></div><div>Age Doubled:<span>72</span></div></div><div>First Name:<span>14</span><input value=\\"14\\"><span>(numRendered: 28)</span></div><div>Last Name:<span>25</span><input value=\\"25\\"><span>(numRendered: 32)</span></div><div><div>Age:<input value=\\"36\\"><span>(numRendered: 36)</span></div><div>Age Doubled:<span>72</span></div></div></div>
  

"
`;
